### DISCUSSION TITLE
Search problems

### DISCUSSION THREAD

**Discussion by vlad-babin** [pymupdf/PyMuPDF#4167] [Looking for help]
Hello!
I have issues with searching in the document.
Here is the script that performs the search and deletes the lines.

```
input_file_path = request.kwargs["inputFilePath"]
search_value = "Reference is made"

doc = pymupdf.open(input_file_path)

for page in doc:
    instances = page.search_for(search_value)
    for inst in instances:
        page.add_redact_annot(inst, fill=(0, 0, 0))
    page.apply_redactions()

doc.save('redacted_document.pdf')
doc.close()
```

The problem is that the search results provide coordinates slightly above the actual line in the file.

This issue occurs in files that had the problem described in this [link](https://github.com/pymupdf/PyMuPDF/issues/3751#issuecomment-2271427952) before version 1.24.14.

I’m attaching the files for reference.
[original.pdf](https://github.com/user-attachments/files/18197483/original.pdf)
[redacted.pdf](https://github.com/user-attachments/files/18197500/redacted.pdf)

This problem appeared in version 1.25.0 (and persists in 1.25.1 as well).

I’m using version 1.25.0.

**Comment by JorjMcKie** [comment]
I found the reason for this weird behavior.
There was a behavior change in MuPDF between the PyMuPDF versions you have observed:
The font binary may (and usually does) contain the float values "ascender" and "descender" which control which portions of the glyphs appear above and below the baseline. Descender is / must be a negative value.
In addition the font object defined in the PDF may contain overriding values for the same. Since recent versions, MuPDF always accepts the PDF-defined values over the font-internal values, because this has proved to deliver a more correct extraction behavior.
In the case of your PDF however, the PDF object font definition contains a **_positive_** descender value. Therefore, the bbox computations are wrong and too much upward by the double value of the font descender.

So I suggest you open an issue - either here or directly at MuPDF's issue system, https://bugs.ghostscript.com/enter_bug.cgi.
If you decide to open a PyMuPDF issue (my recommendation), I will forward it to MuPDF. Please do not forget to attach again your original PDF again.
Thank you.
